includes/polling/ports.inc.php

```diff
@@ -274,8 +288,41 @@
 }//end if
 
 if ($config['enable_ports_poe']) {
-    $port_stats = snmpwalk_cache_oid($device, 'pethPsePortEntry', $port_stats, 'POWER-ETHERNET-MIB');
-    $port_stats = snmpwalk_cache_oid($device, 'cpeExtPsePortEntry', $port_stats, 'CISCO-POWER-ETHERNET-EXT-MIB');
+    // Code by OS device
+    if ($device['os'] == 'procurve') {
+        // Not tested
+        $port_stats_poe = snmpwalk_cache_oid($device, 'pethPsePortEntry', array(), 'POWER-ETHERNET-MIB');
+    }
+
+    if ($device['os'] == 'ios') {
+        $port_stats_poe = snmpwalk_cache_oid($device, 'pethPsePortEntry', array(), 'POWER-ETHERNET-MIB');
+        $port_stats_poe = snmpwalk_cache_oid($device, 'cpeExtPsePortEntry', $port_stats_poe, 'CISCO-POWER-ETHERNET-EXT-MIB');
+        $port_ent_to_if = snmpwalk_cache_oid($device, 'portIfIndex', array(), 'CISCO-STACK-MIB');
+        foreach ($port_ent_to_if as $p_index => $p_stats) {
+            if ($p_stats[portIfIndex] > 0) {
```

Comment:
Can you update to use `'portifIndex'` + any other indexes in arrays which don't contain `''`

Reply:
On the way.